docker/types/services.py

```diff
@@ -319,6 +322,26 @@ def __init__(self, cpu_limit=None, mem_limit=None, cpu_reservation=None,
             reservation['NanoCPUs'] = cpu_reservation
         if mem_reservation is not None:
             reservation['MemoryBytes'] = mem_reservation
+        if generic_reservations is not None:
+            generic_resources = []
```

Comment:
I'd like to allow users to provide this argument in list form (i.e. already parsed) as well ; we just need to have a `isinstance(generic_reservations, list)` that bypasses parsing.

This is useful for people grabbing the resources data by inspecting another service, for example.

Reply:
So just allow:

```
[  {'DiscreteResourceSpec': {'Kind': 'gpu', 'Value': 1}}, ...]
```

I was wondering if we wanted to expose that API to users...

because I am building a CLI app that mimics docker so they are coming in like:
```
gpu=1
```
but I don't think we should use that at the API level?

Reply:
Yeah, CLI-like notation is out of scope for this project ; exposing the original API format when possible is something we do because it allows people to reuse `inspect` results and gives them a workaround when new things make it into the API until we can implement them at the SDK level.

Reply:
ok! Ill change it to match the docker API